lib/Reader/readerir.cpp

```diff
@@ -2020,6 +2020,91 @@ bool GenIR::isUnmanagedPointerType(llvm::Type *Type) {
   return Type->isPointerTy() && !isManagedPointerType(Type);
 }
 
+bool GenIR::isArrayType(llvm::Type *ArrayTy) {
```

Comment:
I was thinking trailing zero-length array fields were rare enough that they were a pretty good indication that this was an array. If we wanted to do more careful checking, I'd probably track the llvm types that were built for arrays explicitly somehow.